perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/UserAttributeProcessor.java

```diff
@@ -18,30 +19,35 @@
 
 	@Autowired
 	protected PerunUser perunUser;
-	
+
 	private static Pattern userSetPattern = Pattern.compile(" set for User:\\[(.*)\\]");
 	private static Pattern userRemovePattern = Pattern.compile(" removed for User:\\[(.*)\\]");
 	private static Pattern userAllAttrsRemovedPattern = Pattern.compile("All attributes removed for User:\\[(.*)\\]");
 
 	//UserExtSources patterns
 	private Pattern addUserExtSourcePattern = Pattern.compile("UserExtSource:\\[(.*)\\] added to User:\\[(.*)\\]");
 	private Pattern removeUserExtSourcePattern = Pattern.compile("UserExtSource:\\[(.*)\\] removed from User:\\[(.*)\\]");
-	
-	
+
+
 	public UserAttributeProcessor() {
 		super(MessageBeans.USER_F, userSetPattern, userRemovePattern, userAllAttrsRemovedPattern);
 	}
-	
-	
+
+
 	public void processAttributeSet(String msg, MessageBeans beans) {
 		// ensure we have the correct beans available
 		if(beans.getAttribute() == null || beans.getUser() == null) {
 			return;
 		}
 		try {
 			log.debug("Setting attribute {} for user {}", beans.getAttribute(), beans.getUser());
-			perunUser.modifyEntry(beans.getUser(), beans.getAttribute());
-		} catch (NamingException | InternalErrorException e) {
+			if (beans.getAttribute().getNamespace().contains(":virt")) {
+				perunUser.modifyEntry(beans.getUser(), ldapcManager.getPerunBl().getAttributesManager().
```

Comment:
Again you can use ``AttributesManagerBl`` just by casting returned object from ``ldapcManager.getPerunBl()`` from ``Perun`` to ``PerunBl``.